diff --git a/configure.ac b/configure.ac
index aad3fa9..bb2fd98 100644
--- configure.ac
+++ configure.ac
@@ -3138,6 +3138,18 @@ $ac_cv_sizeof_long)
   guint32_format='"lu"'
   ;;
 esac
+AS_IF([test x$glib_have_carbon = xyes],
+[
+  gint64='long long'
+  if test -n "$glib_cv_long_long_format"; then
+    gint64_modifier='"'$glib_cv_long_long_format'"'
+    gint64_format='"'$glib_cv_long_long_format'i"'
+    guint64_format='"'$glib_cv_long_long_format'u"'
+  fi
+  glib_extension='G_GNUC_EXTENSION '
+  gint64_constant='(G_GNUC_EXTENSION (val##LL))'
+  guint64_constant='(G_GNUC_EXTENSION (val##ULL))'
+], [
 case 8 in
 $ac_cv_sizeof_int)
   gint64=int
@@ -3180,6 +3192,7 @@ $ac_cv_sizeof___int64)
   guint64_constant='(val##ui64)'
   ;;
 esac
+])
 glib_size_t=$ac_cv_sizeof_size_t
 glib_ssize_t=$ac_cv_sizeof_ssize_t
 glib_size_type_define="$glib_size_type"
 