--- lib/Finance/Quote/YahooWeb.pm
+++ lib/Finance/Quote/YahooWeb.pm
@@ -90,7 +90,8 @@
         my $te = HTML::TableExtract->new(
             headers => ['Date', 'Open', 'High', 'Low', 'Close\*', 'Adj Close\*\*', 'Volume'],
             attribs => { 'data-test' => "historical-prices" } );
-        unless ($te->parse($reply->decoded_content)) {
+        unless ($te->parse($reply->decoded_content) &&
+                defined $te->first_table_found()) {
             $info{ $symbol, "success" } = 0;
             $info{ $symbol, "errormsg" } = "YahooWeb - History table not found.";
             next;
